### PR TITLE
fix(crypto): improve handling of `length` option

### LINKED ISSUE #1429: docs: document about which algorithms support `length` option for stdCrypto.digest
related: https://github.com/denoland/deno_std/pull/1386

### PR DESCRIPTION
**Pull request by iuioiua**
Closes #1429

### CODE REVIEW

**Comment by iuioiua** [comment]
PTAL, @kt3k

File: crypto/crypto.ts [hunk 1]
```diff
@@ -33,7 +36,7 @@
  * - `BLAKE2B-256`
  * - `BLAKE2B-384`
  * - `BLAKE2S`
- * - `BLAKE3`
+ * - `BLAKE3` (length-adjustable)
```
Comment (kt3k) [inline crypto/crypto.ts:39]: What's the difference of `(length-adjustable)` and `(length-entendable)`?

File: crypto/crypto.ts [hunk 2]
```diff
@@ -215,6 +218,11 @@ const stdCrypto: StdCrypto = ((x) => x)({
         // and the data is a single buffer,
         isBufferSource(data)
       ) {
+        if (length !== undefined) {
+          throw new TypeError(
```
Comment (kt3k) [inline crypto/crypto.ts:222]: The native `crypto.subtle.digest` seems ignoring `length` for the case where `length` is irrelevant. Throwing here feels inconsistent with it.

Comment (iuioiua) [reply crypto/crypto.ts]: Good point. I've reverted and instead added a note saying that `length` will be ignored in some cases.

File: crypto/crypto.ts (hunk 1 above)
Comment (iuioiua) [reply crypto/crypto.ts:39]: > "length-extendable" refers to a specific vulnerability in some hash functions, while "length-adjustable" is not a standard term in cryptography but could refer to the general capability of hash functions to process inputs of different lengths.

-- ChatGPT

**Review by littledivy — APPROVED** [review]
LGTM

**Pull request merged**